src/main/java/org/apache/sysml/runtime/instructions/cp/ParamservBuiltinCPInstruction.java

```diff
@@ -410,24 +425,11 @@ private void disjointContiguous(MatrixObject features, MatrixObject labels, Matr
 		}
 
 		// validation data
-		List<MatrixObject> pvfs = disjointContiguous(workers.size(), valFeatures);
-		List<MatrixObject> pvls = disjointContiguous(workers.size(), valLabels);
+		List<MatrixObject> pvfs = dp.doPartition(workers.size(), valFeatures);
+		List<MatrixObject> pvls = dp.doPartition(workers.size(), valLabels);
```

Comment:
Sorry for not catching this before - we don't need to partition the validation data at all. All workers should get the same read-only validation data.